Sources/CommandLine.swift

```
@@ -198,6 +198,8 @@ func printHelp(as type: CLI.OutputType) {
     --lenient          Suppress errors for unformatted code in --lint mode
     --verbose          Display detailed formatting output and warnings/errors
     --quiet            Disables non-critical output messages and warnings
+    --reporter         Defines a custom reporter (only `json` is supported)
+    --reporter-output  Reporter output file (defaults to `output.<reporter>`)
```

The `--reporter-output` parameter seems redundant - we can just re-use the existing `--output` parameter.

Also, the output shouldn't default to a hard-coded filename if no output file is specified, it should just write the json to stdout.

---

Isn't `--output` the output of the formatted file? reusing it here would be a bit confusing 🤔 
```
    --output           Output path for formatted file(s) (defaults to input path)
```

What about having just a `--report` parameter, and determine the type from the provided file extension?

---

Replaced `reporter` and `reporter-output` with single `report` parameter (from which also the type can be inferred in the future.
Please let me know whether it is fine 🙌

---

The problem with that is that if you don't specify an --output file then there's no way for it to infer what format should be sent to stdout.

---

The `--report` parameters tales in input the report file, not the type, so you can use it as:
```
swiftformat <whatever option> --report report.json
```

So when printing the json report the rest of the output and the parameters are not affected.

What do you think?

---

@nicklockwood did you had time to have a look at this?
Not urgent at all, just wanted to be sure you didn't miss my message 🙌

---

Hmm, I was under the impression that reporting was only applicable when running in --lint mode, in which case --output is unused, so having --report take a separate path seemed redundant. But I now realize that you're intending for report to work even when formatting? In which case I agree a separate --report path makes sense.

---

For my use case I would use it only for linting, but I think having a separate parameter may be more generic and cover more use cases 🙌